plone/app/contentlisting/catalog.py

```diff
@@ -39,12 +39,13 @@ def __getattr__(self, name):
 
         if name.startswith('_'):
             raise AttributeError(name)
-        if hasattr(aq_base(self._brain), name):
-            return getattr(self._brain, name)
-        elif hasattr(aq_base(self.getObject()), name):
-            return getattr(aq_base(self.getObject()), name)
-        else:
-            raise AttributeError(name)
+        brain_name = getattr(aq_base(self._brain), name, None)
+        if brain_name is not None:
+            return brain_name
+        object_name = getattr(aq_base(self.getObject()), name, None)
```

Comment:
You need to use a _marker object_ instead of `None` here – this goes for both your `brain_name` and `object_name`.

As an example, `object()` (on module level) would be a good marker object.